### PR TITLE
Log successful API Authentication

### PR DESCRIPTION
**Pull request by colinhoglund**
Adding a debug log line that indicates when a user successfully authenticates with the API. This provides useful context about which users are making API calls successfully.

### CODE REVIEW

File: handler/api/auth/auth.go [hunk 1]
```diff
@@ -47,6 +47,9 @@ func HandleAuthentication(session core.Session) func(http.Handler) http.Handler
 				log = log.WithField("user.admin", user.Admin)
 			}
 			log = log.WithField("user.login", user.Login)
+
+			log.Debugln("api: authentication successful")
+
```
Comment (whwalter) [inline handler/api/auth/auth.go:52]: It doesn't seem like we know if this was token auth or a session auth here.  [session.Get](https://github.com/harness/drone/blob/bafaa40ded92c03a3a736b4b9e85081f920599ba/session/session.go#L70) will try all the auth paths and get a user from token or session auth.

Comment (colinhoglund) [reply handler/api/auth/auth.go:52]: This is only adding a new log event on successful API authentication. Additional log fields would be a slightly larger change that would require some refactoring.

**Comment by colinhoglund** [comment]
On second thought, this doesn't appear to include the request-id, so it's probably not particularly valuable as is.

File: handler/api/auth/auth.go (hunk 1 above)
Comment (whwalter) [reply handler/api/auth/auth.go:52]: My point is that the drone server UI functions like an SPA and makes client side api requests via session cookie. 

As an example, when I load a repository in the drone UI, my browser makes the request:

```
https://drone.corp.mongodb.com/api/repos/{organization}/{repository}/builds?page=1&per_page=50
```

Which gets logged regardless of whether this was api bearer token/token param authentication or session based authentication.

**Pull request closed without merging**